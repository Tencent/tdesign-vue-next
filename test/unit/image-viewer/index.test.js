import { mount } from '@vue/test-utils';
import { ImageViewer } from '@/src/image-viewer/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Image-viewer', () => {
  // test props api
  describe(':props', () => {
    it('images, index, visible', () => {
      const images = [
        'https://tdesign.gtimg.com/demo/demo-image-1.png',
        'https://tdesign.gtimg.com/demo/demo-image-2.png',
        'https://tdesign.gtimg.com/demo/demo-image-3.png',
      ];
      const wrapper = mount(ImageViewer, {
        props: {
          index: 1,
          visible: true,
          images,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('showOverlay', () => {
      const wrapper = mount(ImageViewer, {
        props: {
          visible: true,
          showOverlay: false,
        },
      });
      expect(wrapper.find('.t-image-viewer__modal--mask').exists()).toBe(false);
    });

    // it('zIndex', () => {
    //   const wrapper = mount(ImageViewer, {
    //     props: {
    //       visible: true,
    //       zIndex: 2
    //     },
    //     attachTo: document.getElementById('#app')
    //   })
    //   expect(document.querySelector('.t-image-viewer-preview-image').attributes.style).toMatch(/z-index: 2/)
    // })

    it('defaultVisible', () => {
      const wrapper = mount(ImageViewer, {
        props: {
          defaultVisible: true,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  // test events
  // describe('@event', () => {});

  // test slots
  // describe('<slot>', () => {
  //   it('', () => {});
  // });

  // test exposure function
  // describe('function', () => {
  //   it('', () => {});
  // });
});
