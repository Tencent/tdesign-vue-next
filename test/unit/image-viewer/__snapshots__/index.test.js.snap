// Vitest Snapshot v1

exports[`Image-viewer > :props > defaultVisible 1`] = `
<div
  data-v-app=""
>
  
  <!---->
  <!--teleport start-->
  <!--teleport end-->
  
</div>
`;

exports[`Image-viewer > :props > images, index, visible 1`] = `
<div
  data-v-app=""
>
  
  <!---->
  <!--teleport start-->
  <!--teleport end-->
  
</div>
`;
